src/dataServices/dynamoDbDataService.test.ts

```diff
@@ -86,3 +96,173 @@ describe('updateResource', () => {
+        outputFormat: 'ndjson',
+        since: '2020-08-01T12:00:00Z',
+        type: 'Patient',
+        groupId: '1',
+    };
+
+    test('Successful initiate export request', async () => {
+        // BUILD
+        // Return an export request that is in-progress
+        AWSMock.mock('DynamoDB', 'query', (params: QueryInput, callback: Function) => {
+            callback(null, {
+                // requesterUserId does not equal requesterUserId from initiateExportRequest therefore we are below throttle limit
+                Items: [DynamoDBConverter.marshall({ requesterUserId: 'userId-2', jobStatus: 'in-progress' })],
+            });
+        });
+        AWSMock.mock('DynamoDB', 'putItem', (params: QueryInput, callback: Function) => {
+            // Successfully update export-request table with request
+            callback(null, {});
+        });
+
+        const dynamoDbDataService = new DynamoDbDataService(new AWS.DynamoDB());
+
+        // OPERATE
+        const jobId = await dynamoDbDataService.initiateExport(initiateExportRequest);
+
+        // CHECK
+        expect(jobId).not.toBeUndefined();
+    });
+
+    test('throttle limit exceeds MAXIMUM_CONCURRENT_REQUEST_PER_USER because user already has an in-progress request', async () => {
+        // BUILD
+        // Return an export request that is in-progress
+        AWSMock.mock('DynamoDB', 'query', (params: QueryInput, callback: Function) => {
+            callback(null, {
+                Items: [DynamoDBConverter.marshall({ jobOwnerId: 'userId-1', jobStatus: 'in-progress' })],
+            });
+        });
+
+        const dynamoDbDataService = new DynamoDbDataService(new AWS.DynamoDB());
+
+        // OPERATE
+        try {
+            await dynamoDbDataService.initiateExport(initiateExportRequest);
+        } catch (e) {
+            // CHECK
+            expect(e).toMatchObject(new TooManyConcurrentExportRequestsError());
+        }
+    });
+
+    test('throttle limit exceeded MAXIMUM_SYSTEM_LEVEL_CONCURRENT_REQUESTS because system already has two in-progress request', async () => {
+        // BUILD
+        // Return two export requests that are in-progress
+        AWSMock.mock('DynamoDB', 'query', (params: QueryInput, callback: Function) => {
+            callback(null, {
+                Items: [
+                    DynamoDBConverter.marshall({ jobOwnerId: 'userId-2', jobStatus: 'in-progress' }),
+                    DynamoDBConverter.marshall({ jobOwnerId: 'userId-3', jobStatus: 'in-progress' }),
+                ],
+            });
+        });
+
+        const dynamoDbDataService = new DynamoDbDataService(new AWS.DynamoDB());
+
+        // OPERATE
+        try {
+            await dynamoDbDataService.initiateExport(initiateExportRequest);
+        } catch (e) {
+            // CHECK
+            expect(e).toMatchObject(new TooManyConcurrentExportRequestsError());
+        }
+    });
+});
+
+describe('cancelExport', () => {

```

Comment:
What if you cancel a job that is already completed?

Reply:
I'm going to add code to throw an error if we're trying to cancel an export job that is in `completed` or `failed` state as a safety precaution in the persistence layer.

However, technically we should already be able to catch this invalid state in the router section. In the router, we currently pull the job status to check if a user has access to the job. In this check, we can also check that the job is not in the state of `completed` or `failed`. If they're in those state, we can send a 400 error to the user. I'll update the router PR to include this logic.

Reply:
We should only be able to cancel 'in-progress' states right?

Reply:
Already handled